src/Core/Scheduler/Providers/TMS_Provider.js

```diff
@@ -0,0 +1,69 @@
+import * as THREE from 'three';
+import OGCWebServiceHelper from './OGCWebServiceHelper';
+import Extent from '../../Geographic/Extent';
+
+function TMS_Provider() {
+}
+
+TMS_Provider.prototype.preprocessDataLayer = function preprocessDataLayer(layer) {
+    if (!layer.extent) {
+        throw new Error(`Missing extent property for layer '${layer.id}'`);
+    }
+    if (!layer.projection) {
+        throw new Error(`Missing projection property for layer '${layer.id}'`);
+    }
+    layer.extent = new Extent(layer.projection, ...layer.extent);
+    if (!layer.options.zoom) {
+        layer.options.zoom = {
+            min: 0,
+            max: 18,
+        };
+    }
+};
+
+TMS_Provider.prototype.url = function url(coTMS, layer) {
+    /* eslint-disable no-template-curly-in-string */
+    return layer.url.replace('${z}', coTMS.zoom)
+        .replace('${y}', coTMS.row)
+        .replace('${x}', coTMS.col);
+    /* eslint-enable no-template-curly-in-string */
+};
+
+TMS_Provider.prototype.executeCommand = function executeCommand(command) {
+    const layer = command.layer;
+    const tile = command.requester;
+    const coordTMS = tile.getCoordsForLayer(layer)[0];
+    const coordTMSParent = (command.targetLevel < coordTMS.zoom) ?
+        OGCWebServiceHelper.WMTS_WGS84Parent(coordTMS, command.targetLevel) :
+        undefined;
```

Comment:
So this is : "if the requested level is less than my current level, then let's find out which one of my parents is on this level". That's it? 

How is it possible that the requester ask for a level lower than its own? For some network strategy like dichotomy?